File: R/tutorial/simple-examples/parameter_flow.R
library(metaflow)

start <- function(self){
    print(paste("alpha is", self$alpha))
}

end <- function(self){
    print(paste("alpha still is", self$alpha))
}

metaflow("ParameterFlow") %>%
    parameter("alpha", 
              help="learning rate", 
              required = TRUE,
              type = "bool") %>%    
    step(step="start", 
         r_function=start, 
         next_step="end") %>%
    step(step="end", 
         r_function=end) %>% 
    run()

File: metaflow-tutorials/parameter_flow.py
from metaflow import FlowSpec, Parameter, step
class ParameterFlow(FlowSpec):
    alpha = Parameter('with',
                      help='Learning rate',
                      default=0.01)
    @step
    def start(self):
        print('alpha is %f' % self.alpha)
        self.next(self.end)
    @step
    def end(self):
        print('alpha is still %f' % self.alpha)
if __name__ == '__main__':
    ParameterFlow()